scheduler/services/resource/ocs_resource_service.py

```diff
@@ -267,7 +289,9 @@ def _add_dates_to_dict(prog_dict: Dict[ProgramID, Set[date]], text: str, add_dat
             date_set = prog_dict.setdefault(prog_id, set())
             date_set.add(add_date)
 
-    def _load_spreadsheet(self) -> NoReturn:
+    def _load_spreadsheet(self, filename: Optional[os.PathLike] = None,
+                          file: Optional[BytesIO] = None,
+                          from_gdrive: Optional[bool] = False) -> NoReturn:
```

Comment:
Why is this `Optional[bool]`? That gives it three states, which I don't think is what we want:
```python
None
Some(True)
Some(False)
```

If I understand correctly, you want to be able to use a `filename`, a stream, or a downloaded file... as it stands, we can get all three here, or some combination of all three, which means we should model it differently.

I am thinking something like:
```
FileName = os.PathLike
StreamData = BytesIO
```
and then we do something like:
```python
def _load_spreadsheet(self, data_source: Optional[Union[os.PathLike, BytesIO]]) -> None: 
```
(Note the return type should be `None`).

Then if `data_source` is:
1. `None`, we download from Google drive.
2. `os.PathLike`, treat as you do below.
3. `BytesIo`, treat as you do below.

Could be a good place to try to use `match` to process.

This is pretty confusing because there are three valid choices, and yet the API presents a combination of 12 choices (any mix of `None` and `Some` for each of the first two parameters, and three choices as explained above for the `Optional[bool]`).

Reply:
You are right. I'll fix it.